spring-integration-http/src/main/java/org/springframework/integration/http/outbound/HttpRequestExecutingMessageHandler.java

```diff
@@ -351,53 +348,13 @@ public String getComponentType() {
 	@Override
 	protected void doInit() {
 		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(this.getBeanFactory());
-
-		ConversionService conversionService = this.getConversionService();
-		if (conversionService == null){
-			conversionService = new GenericConversionService();
-		}
-		if (conversionService instanceof ConverterRegistry){
-			ConverterRegistry converterRegistry =
-					(ConverterRegistry) conversionService;
-
-			converterRegistry.addConverter(new ClassToStringConverter());
-			converterRegistry.addConverter(new ObjectToStringConverter());
-
-			this.evaluationContext.setTypeConverter(new StandardTypeConverter(conversionService));
-		}
-		else {
-			logger.warn("ConversionService is not an instance of ConverterRegistry therefore" +
-					"ClassToStringConverter and ObjectToStringConverter will not be registered");
-		}
-	}
-
-	private class ClassToStringConverter implements Converter<Class<?>, String> {
-		@Override
-		public String convert(Class<?> source) {
-			return source.getName();
-		}
-	}
-
-	/**
-	 *  Spring 3.0.7.RELEASE unfortunately does not trigger the ClassToStringConverter.
-	 *  Therefore, this converter will also test for Class instances and do a
-	 *  respective type conversion.
-	 *
-	 */
-	private class ObjectToStringConverter implements Converter<Object, String> {
-		@Override
-		public String convert(Object source) {
-			if (source instanceof Class) {
-				return ((Class<?>) source).getName();
-			}
-			return source.toString();
-		}
 	}
 
 	@Override
 	protected Object handleRequestMessage(Message<?> requestMessage) {
-		String uri = this.uriExpression.getValue(this.evaluationContext, requestMessage, String.class);
-		Assert.notNull(uri, "URI Expression evaluation cannot result in null");
+		Object uri = this.uriExpression.getValue(this.evaluationContext, requestMessage);
+		Assert.state(uri instanceof String || uri instanceof URI,
+				"'uriExpression' evaluation must result in 'String' or 'URI' instance, but not in the: " + uri);
```

Comment:
We needed `ClassToStringConverter` before because `determineExpectedResponseType` got deal just only with `String`. But after introduction support for `ParameterizedTypeReference`, I reworked that to the raw `Object` and assert for result type.
This conversion was fully based (and can be now) on the `ObjectToStringConverter`.
We don't get deal here with `Class<?>` we just need a proper instance from expression evaluation.

Not sure why you mention here the `Class.toString()` concern.
My idea just to bypass `ConversionService` and keep in mind in the future that it isn't good idea to register `Converter`s unconditionally from our components.